### ISSUE TITLE
zfile_output creates directory if full path contains no slashes

### ISSUE THREAD

**Issue by detly** [zeromq/czmq#2208]
[In `zfile_output()`](https://github.com/zeromq/czmq/blob/master/src/zfile.c#L435), if `self->fullname` is simply a file with no path eg. `temp.txt`, then `last_slash` will be `NULL`. But there is no check for this before creating the "containing" directory ie. `zsys_dir_create (file_path)` simply creates *a directory* named `temp.txt`. This was not behaviour I expected.

**Comment by sphaero** [comment]
Indeed, we see the same problem

Here's some code to reproduce
```
    // create a file without a path
    zfile_t *relfile = zfile_new(NULL, "relfile.tst");
    assert(relfile);
    int rco = zfile_output(relfile);
    assert(rco == 0);
```

**Comment by sphaero** [comment]
@detly, can you verify this fix and merge the PR?

**Comment by sphaero** [comment]
fixed by #2211

**Comment by detly** [comment]
Sorry for the delay. Yes, I can confirm this fixes the issue I saw in my actual code base.
